YAMLConfiguration apiVersion is considered float causing issues when reading it as String

```
java.lang.RuntimeException: Error in retrieving configuration property spline.lineageDispatcher.logging.apiVersion
	at za.co.absa.commons.config.ConfigurationImplicits$ConfigurationRequiredWrapper$$anonfun$getRequired$extension$2.applyOrElse(ConfigurationImplicits.scala:127)
	at za.co.absa.commons.config.ConfigurationImplicits$ConfigurationRequiredWrapper$$anonfun$getRequired$extension$2.applyOrElse(ConfigurationImplicits.scala:122)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
	at scala.util.Failure.recover(Try.scala:234)
	at za.co.absa.commons.config.ConfigurationImplicits$ConfigurationRequiredWrapper$.getRequired$extension(ConfigurationImplicits.scala:122)
	at za.co.absa.commons.config.ConfigurationImplicits$ConfigurationRequiredWrapper$.$anonfun$getRequiredString$3(ConfigurationImplicits.scala:54)
	at za.co.absa.spline.harvester.dispatcher.LoggingLineageDispatcher.<init>(LoggingLineageDispatcher.scala:33)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at za.co.absa.commons.HierarchicalObjectFactory$$anonfun$instantiate$4.applyOrElse(HierarchicalObjectFactory.scala:50)
	at za.co.absa.commons.HierarchicalObjectFactory$$anonfun$instantiate$4.applyOrElse(HierarchicalObjectFactory.scala:49)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
	at scala.util.Failure.recover(Try.scala:234)
	at za.co.absa.commons.HierarchicalObjectFactory.instantiate(HierarchicalObjectFactory.scala:49)
	at za.co.absa.spline.agent.AgentBOM$$anon$1.obtain(AgentBOM.scala:119)
	at za.co.absa.spline.agent.AgentBOM$$anon$1.$anonfun$lineageDispatcher$2(AgentBOM.scala:100)
	at scala.collection.immutable.List.map(List.scala:293)
	at za.co.absa.spline.agent.AgentBOM$$anon$1.lineageDispatcher$lzycompute(AgentBOM.scala:100)
	at za.co.absa.spline.agent.AgentBOM$$anon$1.lineageDispatcher(AgentBOM.scala:91)
	at za.co.absa.spline.harvester.SparkLineageInitializer.$anonfun$createListener$8(SparkLineageInitializer.scala:146)
	at org.apache.spark.internal.Logging.logInfo(Logging.scala:57)
	at org.apache.spark.internal.Logging.logInfo$(Logging.scala:56)
	at za.co.absa.spline.harvester.SparkLineageInitializer.logInfo(SparkLineageInitializer.scala:119)
	at za.co.absa.spline.harvester.SparkLineageInitializer.createListener(SparkLineageInitializer.scala:146)
	at za.co.absa.spline.harvester.SparkLineageInitializer.$anonfun$createListener$7(SparkLineageInitializer.scala:141)
	at za.co.absa.spline.harvester.SparkLineageInitializer.assureOneListenerPerSession(SparkLineageInitializer.scala:207)
	at za.co.absa.spline.harvester.SparkLineageInitializer.$anonfun$createListener$6(SparkLineageInitializer.scala:141)
	at za.co.absa.spline.harvester.SparkLineageInitializer.withErrorHandling(SparkLineageInitializer.scala:176)
	at za.co.absa.spline.harvester.SparkLineageInitializer.createListener(SparkLineageInitializer.scala:138)
	at za.co.absa.spline.harvester.SparkLineageInitializer$SplineSparkSessionWrapper.enableLineageTracking(SparkLineageInitializer.scala:106)
	at za.co.absa.spline.issue.DeltaMergeDSV2Job$.delayedEndpoint$za$co$absa$spline$issue$DeltaMergeDSV2Job$1(DeltaMergeDSV2Job.scala:36)
	at za.co.absa.spline.issue.DeltaMergeDSV2Job$delayedInit$body.apply(DeltaMergeDSV2Job.scala:25)
	at scala.Function0.apply$mcV$sp(Function0.scala:39)
	at scala.Function0.apply$mcV$sp$(Function0.scala:39)
	at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:17)
	at scala.App.$anonfun$main$1$adapted(App.scala:80)
	at scala.collection.immutable.List.foreach(List.scala:431)
	at scala.App.main(App.scala:80)
	at scala.App.main$(App.scala:78)
	at za.co.absa.spline.SparkApp.main(SparkApp.scala:27)
	at za.co.absa.spline.issue.DeltaMergeDSV2Job.main(DeltaMergeDSV2Job.scala)
Caused by: org.apache.commons.configuration.ConversionException: 'apiVersion' doesn't map to a String object
	at org.apache.commons.configuration.AbstractConfiguration.getString(AbstractConfiguration.java:1047)
	at org.apache.commons.configuration.AbstractConfiguration.getString(AbstractConfiguration.java:1018)
	at za.co.absa.commons.config.ConfigurationImplicits$ConfigurationRequiredWrapper$.$anonfun$getRequiredString$1(ConfigurationImplicits.scala:54)
	at za.co.absa.commons.config.ConfigurationImplicits$ConfigurationRequiredWrapper$.$anonfun$getRequired$extension$1(ConfigurationImplicits.scala:120)
	at scala.util.Try$.apply(Try.scala:213)
	at za.co.absa.commons.config.ConfigurationImplicits$ConfigurationRequiredWrapper$.getRequired$extension(ConfigurationImplicits.scala:120)
	... 38 more
```

--- a/core/src/main/resources/spline.default.yaml
+++ b/core/src/main/resources/spline.default.yaml
@@ -78,7 +78,7 @@ spline:
         kafka:
             className: za.co.absa.spline.harvester.dispatcher.KafkaLineageDispatcher
             # Spline Producer API version
-            apiVersion: 1.2
+            apiVersion: "1.2"
             # producer configs as defined by kafka (bootstrap.servers, key.serializer, etc) all kafka configs are supported
             producer.bootstrap.servers:
             producer.key.serializer: org.apache.kafka.common.serialization.StringSerializer
@@ -145,7 +145,7 @@ spline:
             disableSslValidation: false
             namespace: default
             # Spline Producer API version
-            apiVersion: 1.2
+            apiVersion: "1.2"
             # Base URL of the Spline Producer REST API endpoint
             api.url:
 

--- a/core/src/main/scala/za/co/absa/spline/harvester/dispatcher/AbstractJsonLineageDispatcher.scala
+++ b/core/src/main/scala/za/co/absa/spline/harvester/dispatcher/AbstractJsonLineageDispatcher.scala
@@ -18,24 +18,30 @@ package za.co.absa.spline.harvester.dispatcher
 
 import za.co.absa.commons.EnumUtils.EnumOps
 import za.co.absa.commons.reflect.EnumerationMacros
-import za.co.absa.spline.harvester.dispatcher.AbstractJsonLineageDispatcher.ModelEntity
-import za.co.absa.spline.producer.model.{ExecutionEvent, ExecutionPlan}
-
+import za.co.absa.spline.harvester.dispatcher.modelmapper.ModelMapper
 import za.co.absa.spline.harvester.json.HarvesterJsonSerDe
+import za.co.absa.spline.producer.model.{ExecutionEvent, ExecutionPlan}
 
 abstract class AbstractJsonLineageDispatcher
   extends LineageDispatcher {
 
-  import ModelEntity.Implicits._
   import HarvesterJsonSerDe.impl._
 
-  final override def send(plan: ExecutionPlan): Unit = send(Array(getEntityName(ModelEntity.Plan), plan).toJson)
+  private val apiVersion = ProducerApiVersion.V1_2
+  private val modelMapper = ModelMapper.forApiVersion(apiVersion)
 
-  final override def send(event: ExecutionEvent): Unit = send(Array(getEntityName(ModelEntity.Event), event).toJson)
+  final override def send(plan: ExecutionPlan): Unit = {
+    val json = modelMapper.toDTO(plan).map(_.toJson).getOrElse("{}")
+    send(s"ExecutionPlan (apiVersion: ${apiVersion.asString}):\n$json")
+  }
+
+  final override def send(event: ExecutionEvent): Unit = {
+    val json = modelMapper.toDTO(event).map(_.toJson).getOrElse("{}")
+    send(s"ExecutionEvent (apiVersion: ${apiVersion.asString}):\n$json")
+  }
 
   protected def send(json: String): Unit
 
-  private def getEntityName(entity: ModelEntity): String = entity.name.toLowerCase
 }
 
 object AbstractJsonLineageDispatcher {

--- a/core/src/test/scala/za/co/absa/spline/harvester/dispatcher/ConsoleLineageDispatcherSpec.scala
+++ b/core/src/test/scala/za/co/absa/spline/harvester/dispatcher/ConsoleLineageDispatcherSpec.scala
@@ -35,7 +35,8 @@ class ConsoleLineageDispatcherSpec
 
   it should "send lineage to the stdout" in {
     assertingStdOut(include(
-    """["event",{"planId":"12345678-90ab-cdef-1234-567890abcdef","labels":{},"timestamp":999,"extra":{}}]""")) {
+      """ |ExecutionEvent (apiVersion: 1.2):
+          |{"planId":"12345678-90ab-cdef-1234-567890abcdef","timestamp":999}""".stripMargin)) {
       new ConsoleLineageDispatcher(new BaseConfiguration {
         addProperty("stream", "OUT")
       }).send(ExecutionEvent(uuid1, Map.empty, 999, None, None, None, Map.empty))
@@ -44,7 +45,8 @@ class ConsoleLineageDispatcherSpec
 
   it should "send lineage to the stderr" in {
     assertingStdErr(include(
-    """["event",{"planId":"12345678-90ab-cdef-1234-567890abcdef","labels":{},"timestamp":999,"extra":{}}]""")) {
+      """ |ExecutionEvent (apiVersion: 1.2):
+          |{"planId":"12345678-90ab-cdef-1234-567890abcdef","timestamp":999}""".stripMargin)) {
       new ConsoleLineageDispatcher(new BaseConfiguration {
         addProperty("stream", "ERR")
       }).send(ExecutionEvent(uuid1, Map.empty, 999, None, None, None, Map.empty))

